### ISSUE TITLE
Types generation error using groups and content relatioship fields

### ISSUE THREAD

**Issue by amendezm** [prismicio/slice-machine#1387]
### Versions

- slice-machine-ui: v2.2.0 
- node: v20.3.0 

### Reproduction

<details open>
<summary>Additional Details</summary>
<br>

![image](https://github.com/prismicio/slice-machine/assets/26444448/0c39b6fe-ff0a-4069-915b-2defa9abab39)

![image](https://github.com/prismicio/slice-machine/assets/26444448/c8072172-6823-46a1-8715-544b46c6c3cb)

![image](https://github.com/prismicio/slice-machine/assets/26444448/6f8a66a6-5b47-48fd-b06b-74e018c0ce01)

</details>

### Steps to reproduce

Just create a group field in a slice and then add a nested content relatioshinp field

### What is expected?

Types should be generated correctly

### What is actually happening?

Type errors

**Comment by xrutayisire** [comment]
Hi,

Good catch, thank you for the issue. To unlock you, can you pin the dependency of `prismic-ts-codegen` and re try to add any field to your group. It should fix the problem.

Here is an example with override:

```
"overrides": {
    "prismic-ts-codegen": "0.1.20"
  }
```

We will release a new Slice Machine and adapter version with the fix beginning of next week 🙌
Don't forget to remove the overrides after the release, I will post here about it.

Thank you,
Xavier

**Comment by amendezm** [comment]
Hi Xavier,

Thank you for the quick response and the solution! I've pinned the dependency of prismic-ts-codegen to version 0.1.20 as suggested, and it resolved the issue. I was able to add fields to my group successfully.

I appreciate the heads-up about the upcoming release. I'll keep an eye out for your update and will remove the overrides once the new versions are available.

Thanks again for your help!

Best regards,

Alexis Mendez

**Comment by amendezm** [comment]
![image](https://github.com/prismicio/slice-machine/assets/26444448/dfe31c36-2d44-41f2-8806-bd2952d90704)

**Comment by angeloashmore** [comment]
Hi @amendezm, thanks for letting us know it worked. 🙂

I looked into the issue and it most likely happens because your Slice Machine adapter needs an update. Could you try the following when you have some time?

1. Remove the `"overrides"` property in `package.json` suggested by @xrutayisire above.
2. Update your project's Slice Machine adapter to the latest version. The package will look like `@slicemachine/adapter-*` (e.g. `@slicemachine/adapter-next`).
3. Re-generate your TypeScript types by starting Slice Machine and editing a slice.

Could you let us know if those steps work for you? Thank you!

I updated our starter projects with the latest adapters so new users shouldn't see the issue.

**Comment by amendezm** [comment]
Hi @angeloashmore , thanks for your suggestions! I’ve already updated to the latest version of the Slice Machine adapter, and it seems to be working fine now.

Just to confirm, I followed these steps:

1. Removed the "overrides" property in package.json.
2. Updated the Slice Machine adapter to the latest version (@slicemachine/adapter-next in my case).
3. Re-generated the TypeScript types by starting Slice Machine and editing a slice.

Everything is now working as expected. Thanks for your help!
